src/components/ha-service-control.ts

```diff
@@ -240,12 +240,24 @@ export class HaServiceControl extends LitElement {
         ...value,
         selector: value.selector as Selector | undefined,
       }));
+
+      const hasSelector: string[] = [];
+      fields.forEach((field) => {
+        if ((field as any).fields) {
```

Comment:
Could not get the typing figured out here so had to avoid it. It seems like we're doing some type hacking in the header to try to make this work, but I think what we probably need is an update to home-assistant-js-websocket to properly type fields with subfields?

Reply:
https://github.com/home-assistant/home-assistant-js-websocket/blob/5157de3d1130646d361a405841fd5a5715dd64ac/lib/types.ts#L100